core/src/test/java/org/fao/geonet/kernel/security/ldap/LDAPUserDetailsContextMapperWithProfileSearchEnhancedTest.java

```diff
@@ -0,0 +1,439 @@
+
+        //make sure its already in the repository, or nothing will happen
+        Group group = new Group().setName("GENERAL");
+        group = groupRepository.save(group);
+
+
+        UserDetails details = ldapUserDetailsService.loadUserByUsername("jgee@example.com");
+        assertNotNull(details);
+
+        LDAPUser user1 = (LDAPUser) details;
+        assertEquals(0, user1.getPrivileges().size());
+
+        User user = userRepository.findOneByUsername("jgee@example.com");
+        assertNotNull(user);
+        assertEquals("gee", user.getSurname());
+        assertEquals(Profile.RegisteredUser, user.getProfile());
+
+        List<UserGroup> ug = userGroupRepository.findAll(UserGroupSpecs.hasUserId(user.getId()));
+        assertNotNull(ug);
+        assertEquals(0, ug.size());
+    }
+
+
+    @Test
+    @Transactional
+    public void test_ldapAuthProvider() {
+        UsernamePasswordAuthenticationToken userpass = new UsernamePasswordAuthenticationToken("dblasby@example.com", "blasby1", null);
+
+        Authentication result = ldapAuthProvider.authenticate(userpass);
+        assertNotNull(result);
+        assertTrue(result.isAuthenticated());
+    }
+
+    @Test(expected = BadCredentialsException.class)
+    public void test_ldapAuthProvider_badPass() {
+        UsernamePasswordAuthenticationToken userpass = new UsernamePasswordAuthenticationToken("dblasby@example.com", "BAD_PASSWORD", null);
+
+        //will throw because the password is wrong
+        Authentication result = ldapAuthProvider.authenticate(userpass);
+    }
+
+
+    //=========================================================================================
+
+    public final static String ldapSearchBase = "dc=example,dc=com";
+
+
+    DirectoryService LDAPservice = null;
+    LdapServer ldapServer = null;
+
+    public void setupLDAP() throws Exception {
+        CreateLdapServer classLdapServerBuilder = this.getClass().getAnnotation(CreateLdapServer.class);
+        CreateDS dsBuilder = this.getClass().getAnnotation(CreateDS.class);
+        LDAPservice = DSAnnotationProcessor.createDS(dsBuilder);
+        ApplyLdifFiles applyLdifFiles = this.getClass().getAnnotation(ApplyLdifFiles.class);
+        DSAnnotationProcessor.injectLdifFiles(applyLdifFiles.clazz(), LDAPservice, applyLdifFiles.value());
+
+        CreateLdapServer createLdapServer = getClass().getAnnotation(CreateLdapServer.class);
+        ldapServer = ServerAnnotationProcessor.instantiateLdapServer(createLdapServer, LDAPservice);
+        ldapServer.start();
+    }
+
+    public void shutdownLDAP() throws Exception {
+        ldapServer.stop();
+        LDAPservice.shutdown();
+        FileUtils.deleteDirectory(LDAPservice.getInstanceLayout().getInstanceDirectory());
+    }
+
+    @After
+    public void after() throws Exception {
+        shutdownLDAP();
+    }
+
+    //gets the spring context
+    @Before
+    public void before() throws Exception {
+        setupLDAP();
```

Comment:
Since you are always using the same LDAP configuration for all test why not start and stop the LDAP server only once using `@BeforeClass` and `@AfterClass` to call `setupLDAP()` and `shutdownLDAP()` methods?

In my machine that reduces the run time of the tests from  1m 17s to 800ms